broadcasting.md

```diff
@@ -208,44 +208,126 @@ window.Echo = new Echo({
+Once Echo is installed, you are ready to create a fresh Echo instance in your applications JavaScript. A great place to do this is at the bottom of the `resources/js/bootstrap.js` file that is included with the Laravel framework. By default, an example Echo configuration is already included in this file; however, the default configuration in the bootstrap.js file is intended for Pusher. You may copy the configuration below to transition your configuration to Ably
 
 ```js
 import Echo from 'laravel-echo';
-import Pusher from 'pusher-js';
-
-window.Pusher = Pusher;
+import * as Ably from 'ably';
 
+window.Ably = Ably;
 window.Echo = new Echo({
-    broadcaster: 'pusher',
-    key: import.meta.env.VITE_ABLY_PUBLIC_KEY,
-    wsHost: 'realtime-pusher.ably.io',
-    wsPort: 443,
-    disableStats: true,
-    encrypted: true,
+    broadcaster: 'ably',
+});
+
+window.Echo.connector.ably.connection.on(stateChange => {
+    if (stateChange.current === 'connected') {
+        console.log('connected to ably server');
+    }
 });
 ```
 
-Note that our Ably Echo configuration references a `VITE_ABLY_PUBLIC_KEY` environment variable. This variable's value should be your Ably public key. Your public key is the portion of your Ably key that occurs before the `:` character.
+You can set custom [clientOptions](https://docs.ably.io/client-lib-development-guide/features/#options) while creating `Echo` instance.
 
+```
+    broadcaster: 'ably',
+    authEndpoint: 'http://www.localhost:8000/broadcasting/auth'
+    // Additional ably specific options - https://docs.ably.io/client-lib-development-guide/features/#options  
+    realtimeHost: 'realtime.ably.com',
+    restHost: 'rest.ably.com',
+    port: '80',
+    echoMessages: true // By default self-echo for published message is false
+```
 Once you have uncommented and adjusted the Echo configuration according to your needs, you may compile your application's assets:
 
 ```shell
 npm run dev
 ```
 
-> **Note**  
-> To learn more about compiling your application's JavaScript assets, please consult the documentation on [Vite](/docs/{{version}}/vite).
+
+> {tip} To learn more about compiling your application's JavaScript assets, please consult the documentation on [Laravel Mix](/docs/{{version}}/mix).
+
+### Additional supported features (official ably client)
+
+**1. Update token expiry (Default 3600, 1 hr)**
+- Update`ABLY_TOKEN_EXPIRY` in `.env` file. 
+- Update `ably` section under `config/broadcasting.php` with `'token_expiry' => env('ABLY_TOKEN_EXPIRY', 3600)`
+
+**2. Modify channel capability**
+- Channel access can be changed as per [Channel Capabilities](https://ably.com/docs/core-features/authentication#capability-operations)
+```php
+  // file - routes/channels.php
+
+  // for private channel (array is returned as a truthy value)
```

Comment:
The sentence "array is returned as a truthy value" is a bit confusing to me, I'm not sure what you're trying to say here.

Reply:
It has to do with returning either `truthy` or `falsy` values. When `truthy` value is returned, channel access is given. So, it doesn't matter if returned value is of type `array` or `object` or `boolean`, all of them are treated as true values. It's the same concept of `Truthy` and `Falsy` values in js -> https://www.sitepoint.com/javascript-truthy-falsy/

Reply:
OK yeah I get that but I still think this could be clearer - maybe it should say:
```suggestion
  // for private channel (If a falsy value is returned then channel access is not permitted for the client)
```

also FWIW empty arrays are falsy in PHP

Reply:
Updated to ->
`// for private channel (Access is allowed for truthy values and denied for falsy values)`